commands/createfixpullrequests.go

```diff
@@ -385,6 +367,88 @@ func (cfp *CreateFixPullRequestsCmd) updatePackageToFixedVersion(fixDetails *uti
+	scanDetails := utils.NewScanDetails(cfp.details.Client(), cfp.details.ServerDetails, cfp.details.Git).
+		SetProject(cfp.details.Project).
+		SetReleasesRepo(cfp.details.ReleasesRepo()).
+		SetXrayGraphScanParams(cfp.details.Watches, cfp.details.ProjectKey).
+		SetMinSeverity(cfp.details.MinSeverityFilter()).
+		SetFixableOnly(cfp.details.FixableOnly()).
+		SetBranch(remoteBranchName)
+	wd, err := os.Getwd()
+	if err != nil {
+		return
+	}
+	res, isMulti, err := cfp.scan(scanDetails, wd)
+	if err != nil {
+		return
+	}
+	targetFixVersionMap, err := cfp.createFixVersionsMap(res, isMulti)
+	if err != nil {
+		return
+	}
+	return utils.FixVersionsMapToMd5Hash(targetFixVersionMap)
+}
+
+func (cfp *CreateFixPullRequestsCmd) checkActivePullRequestByBranchName(branchName string) (exists bool, err error) {
+	list, err := cfp.details.Client().ListOpenPullRequests(context.Background(), cfp.details.RepoOwner, cfp.details.RepoName)
+	if err != nil {
+		return
+	}
+	for _, pr := range list {
+		if pr.Source.Name == branchName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (cfp *CreateFixPullRequestsCmd) aggregateFixAndOpenPullRequest(fixVersionsMap map[string]*utils.FixDetails, aggregatedFixBranchName string) (err error) {
+	var errList strings.Builder
+	var atLeastOneFix bool
+	log.Info("-----------------------------------------------------------------")
+	log.Info("Starting aggregated dependencies fix")
+	log.Debug("Creating branch", aggregatedFixBranchName, "...")
+	if err = cfp.gitManager.CreateBranchAndCheckout(aggregatedFixBranchName); err != nil {
+		return
+	}
+	// Fix all packages in the same branch if expected error accrued, log and continue.
+	for _, fixDetails := range fixVersionsMap {
+		if err := cfp.updatePackageToFixedVersion(fixDetails); err != nil {
+			cfp.handleUpdatePackageErrors(err, errList)
+		} else {
+			log.Info(fmt.Sprintf("Updated dependency '%s' to version '%s'", fixDetails.ImpactedDependency, fixDetails.FixVersion))
+			atLeastOneFix = true
+		}
+	}
+	if atLeastOneFix {
+		if err = cfp.openAggregatedPullRequest(aggregatedFixBranchName); err != nil {
+			return fmt.Errorf("failed while creating aggreagted pull request. Error: \n%s", err.Error())
+		}
+	}
+	logAppendedErrorsIfExists(errList)
+	log.Info("-----------------------------------------------------------------")
+	return
+}
+
+// Compare scan results by MD5 hashing the resulted fixVersionMap of the current branch and the target branch
+func (cfp *CreateFixPullRequestsCmd) compareScanResults(fixVersionsMap map[string]*utils.FixDetails, aggregatedFixBranchName string) (identical bool, err error) {
+	currentScanHash, err := utils.FixVersionsMapToMd5Hash(fixVersionsMap)
+	if err != nil {
+		return
+	}
+	remoteBranchScanHash, err := cfp.getRemoteBranchScanHash(aggregatedFixBranchName)
+	if err != nil {
+		return
+	}
+	if currentScanHash == remoteBranchScanHash {
+		log.Info("Scan results hasn't change since last Frogbot run...")
```

Comment:
* Bad grammar. 
* The "..." at the end should be replaced with "." because this message does not describe an action in progress.